pyccel/codegen/printing/ccode.py

```diff
@@ -587,6 +603,67 @@ def function_signature(self, expr):
         else:
             return '{0}{1}({2})'.format(ret_type, name, arg_code)
 
+    def _print_IndexedElement(self, expr):
+        if isinstance(expr.base, IndexedVariable):
+            base = expr.base.internal_variable
+        else:
+            base = expr.base
+        inds = list(expr.indices)
+        inds = inds[::-1]
+        base_shape = base.shape
+        allow_negative_indexes = (isinstance(base, IndexedVariable) and \
+                base.internal_variable.allows_negative_indexes)
+        base = self._print(expr.base.label)
+        for i, ind in enumerate(inds):
+            if isinstance(ind, PyccelUnarySub) and isinstance(ind.args[0], LiteralInteger):
+                inds[i] = PyccelMinus(base_shape[i], ind.args[0])
+            else:
+                if isinstance(ind, Slice):
+                    #setting the slice start and end to their correct value if none is provided
+                    start = ind.start
+                    end = ind.end
+                    if ind.start is None:
+                        start = 0
+                    if ind.end is None:
+                        end = expr.base.shape[1]
+                    inds[i] = Slice(start, end)
+                #indices of indexedElement of len==1 shouldn't be a Tuple
+                if isinstance(ind, Tuple) and len(ind) == 1:
+                    inds[i].args = ind[0]
+                if allow_negative_indexes and not isinstance(ind, LiteralInteger):
+                    inds[i] = PyccelMod(ind, base_shape[i])
+        inds = [self._print(i) for i in inds]
+        #set dtype to the C struct types
+        dtype = self._print(expr.dtype)
+        dtype = self.find_in_dtype_registry(dtype, expr.precision, array=True)
+        if expr.rank > 0:
+            return "array_slicing(%s, %s)" % (base, ", ".join(inds))
+        return "%s.%s[get_index(%s, %s)]" % (base, dtype, base, ", ".join(inds))
```

Comment:
Are we assuming that we are dealing with an array and never a tuple here?

Reply:
i assumed that since we are not handling the tuples yet.